src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,360 @@
+            throw new ArrayIndexOutOfBoundsException();
+        }
+
+        int indexToBeDeleted = Integer.parseInt(userInputs[1]) - 1;
+        Task deletedTask = userInputList.remove(indexToBeDeleted);
+
+        //Modify local file
+        writeListToFile();
+
+        //Notify user
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("    " + deletedTask);
+        System.out.println("Now you have " + userInputList.size() + " tasks in the list");
+    }
+
+    public static void markTaskAsDone(String[] userInputs) throws ArrayIndexOutOfBoundsException, IOException {
+        //Check for only 'done'
+        if (userInputs.length == 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+
+        //List indexed from 0, offset by 1
+        int listNumber = Integer.parseInt(userInputs[1]) - 1;
+        userInputList.get(listNumber).setIsDone(true);
+
+        //Modify local file
+        writeListToFile();
+
+        //notify user
+        System.out.println("Nice! I've marked this task as done:\n    " + userInputList.get(listNumber));
+    }
+
+    public static void listTasks() {
+        System.out.println("Here are the tasks in your list:");
+        int counter = 1;
+        for (Task task : userInputList) {
+            System.out.println(counter + ". " + task);
+            counter++;
+        }
+    }
+
+    public static void introMessage() {
+        //Intro message
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm duke.Duke");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    //Split at delimiter with format -> before (delimiter: after)
+    public static String getFormattedString(String[] userInputs, String delimiter) {
+        String text = "";
+        for (int i = 1; i < userInputs.length; i++) {
+            if (userInputs[i].equals(delimiter)) { //split at delimiter
+                switch (delimiter){
+                case "/by":
+                    text += "(by: ";
+                    break;
+                case "/at":
+                    text += "(at: ";
+                    break;
+                default:
+                    text += delimiter;
+                }
+            } else if (i == userInputs.length - 1) { //last word
+                text += userInputs[i] + ")";
+            } else {
+                text += userInputs[i] + " ";
+            }
+        }
+        return text;
+    }
+
+    public static void exitDuke() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+        System.exit(0);
+    }
+
+    //Not implemented
```

Comment:
What is not implemented? (looks like a stray comment, please remove such occurrences)